airbyte-lib/airbyte_lib/file_writers/base.py

```diff
@@ -0,0 +1,70 @@
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+
+"""Define abstract base class for File Writers, which write and read from file storage."""
+
+from __future__ import annotations
+
+import abc
+from dataclasses import dataclass, field
+from typing import TYPE_CHECKING, final
+
+from airbyte_lib.config import CacheConfigBase
+from airbyte_lib.processors import BatchHandle, RecordProcessor
+from overrides import overrides
+
+if TYPE_CHECKING:
+    from pathlib import Path
+
+    import pyarrow as pa
+
+
+DEFAULT_BATCH_SIZE = 10000
+
+
+# The batch handle for file writers is a list of Path objects.
+@dataclass
```

Comment:
There is no logic to clean up the files for the batches after the sync is finished - IMHO we should make sure that after a successful sync, no temporary stuff is left behind

Reply:
Makes sense! 👍

Reply:
Added cleanup steps for deleting files.